### PR TITLE
Proxy multiple Endpoints services

### PR DESCRIPTION
**Pull request by halvards**
This change allows an ESP instance to proxy multiple Endpoints
services, as documented in the commit.

The purpose is to simplify infrastructure setup for providers
exposing multiple APIs.

### CODE REVIEW

File: doc/multiple-services.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        --rollout_strategy=managed
+
+This command will generate two service configuration files: 
+`svc1.example.com_server_config.txt` and `svc2.example.com_server_config.txt`.
+You can refer to these files in the `server_config` elements in the
+`endpoints` blocks in your `nginx.conf` template.
+
+You can use either `start_esp/nginx-auto.conf.template` or
+`docker/custom/nginx.conf.template` as a starting point for your own
+`nginx.conf` template.
```
Comment (qiwzhang) [inline doc/multiple-services.md]: You may want to mention the server_config file name that has to match with ones generated by start_esp

**Comment by qiwzhang** [comment]
The code looks good.   Could you re-base your code?  After that,  I will trigger an e2e test on your PR.
If it passed, I will merge it.

**Comment by qiwzhang** [comment]
@lei-tang could you help test this PR with presubmit test, specially //start_esp/test/  test since it makes changes to start_esp.py?  

Since you are the on-call next week,  I will be off next week. Make sure it passes presubmit test and pr-e2e tests.

Thanks

**Comment by lei-tang** [comment]
@qiwzhang Yes, I will help test this PR.

File: start_esp/start_esp.py [hunk 2]
```diff
@@ -317,21 +324,29 @@ def fetch_service_config(args):
...
+                for idx, service in enumerate(services):
+                    logging.info(
+                        "Fetching the service config ID from the rollouts service")
+                    rollout = fetch.fetch_latest_rollout(args.management,
+                                                         service, token)
+                    args.rollout_ids.append(rollout["rolloutId"])
+                    for version, percentage in rollout["trafficPercentStrategy"]["percentages"].iteritems():
+                        filename = generate_service_config_filename(version)
+                        fetch_and_save_service_config(args.management, service, args.config_dir, token, version, filename)
+                        args.service_config_sets.append({})
```
Comment (qiwzhang) [inline start_esp/start_esp.py]: This append() should be outside of "for version" loop, isn't it?

**Comment by qiwzhang** [comment]
@lei-tang This PR just needs one small change.  Once it is updated, you need to manually start a presubmit test for it.  Once it is passed, merge it. 

Please wait for it in your next week release.

**Comment by halvards** [comment]
You're right, well spotted! I've just pushed a fix.

**Comment by lei-tang** [comment]
I have manually started a presubmit test at https://endpoints-jenkins.appspot.com/job/esp/job/presubmit/1152/ for the SHA 6e193048dbc7977daef75cd0010b9a732816e31f.

**Comment by lei-tang** [comment]
I have also started a pr-e2e test at https://endpoints-jenkins.appspot.com/job/jenkins-testing/job/pr-e2e/142/ for the SHA 6e193048dbc7977daef75cd0010b9a732816e31f.

**Pull request merged**